### PR TITLE
Always export all columns, unsorted and unfiltered

### PR DESCRIPTION
**Pull request by Randy424**
Regarding: [ACM-13463](https://issues.redhat.com/browse/ACM-13463)

### CODE REVIEW

File: frontend/src/ui-components/AcmTable/AcmTable.tsx [hunk 1]
```diff
@@ -841,7 +842,17 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
       }
       return tableItem
     })
-    return { tableItems, totalCount: (isPreProcessed && resultCounts?.itemCount) || tableItems.length }
+    // allTableItems is the complete set of table items used for export
+    const allTableItems: ITableItem<T>[] = items.map((item) => {
```
Comment (Randy424) [inline frontend/src/ui-components/AcmTable/AcmTable.tsx:846]: From Kevin: this logic is being used twice, once for each set of tableItems. If we setup the tableItem array first then apply filters, we could avoid this redundancy.

File: frontend/src/ui-components/AcmTable/AcmTable.tsx [hunk 2]
```diff
@@ -486,12 +486,12 @@ const findFilterMatch = <T, S>(filter: string, filterArray: TableFilterBase<T, S
...
 const applyFilters = <T, S>(
-  items: T[],
+  items: ITableItem<T>[],
   filterSelections: CurrentFilters<S>,
   filterArray: TableFilterBase<T, S>[]
-): T[] => {
+): ITableItem<T>[] => {
   const filterCategories = Object.keys(filterSelections)
-  return items.filter((item: T) =>
+  return items.filter(({ item }) =>
```
Comment (Randy424) [inline frontend/src/ui-components/AcmTable/AcmTable.tsx:494]: Refactored applyFilters to be used with ITableItem array.

File: frontend/src/ui-components/AcmTable/AcmTable.tsx [hunk 3]
```diff
@@ -795,34 +795,15 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
...
-      activeAdvancedFilters.forEach(({ columnId, value, operator }) => {
-        if (columnId && value && operator) {
-          advancedFilterSelections[columnId] = { operator, value }
-        }
-      })
-
-      if (advancedFilters.length && Object.keys(advancedFilterSelections).length) {
-        filteredItems = applyFilters(filteredItems, advancedFilterSelections, advancedFilters)
-      }
-    }
```
Comment (Randy424) [inline frontend/src/ui-components/AcmTable/AcmTable.tsx:823]: The diff picked up primarily on the migration of this code to lines 825+
The code building the tableItem object array now precedes this code.

**Comment by KevinFCormier** [comment]
Great work!

**Pull request merged**